controller/binaries.py

```diff
@@ -1,66 +1,57 @@
 import glob
 from pathlib import Path
+from typing import Any, Dict
+
 from termcolor import colored
 
 ROOT_DIR = Path(__file__).parent.parent.resolve()
-FIRMWARES = {
+FIRMWARES: Dict[str, list] = {
     "T1": [],
     "TT": [],
 }
-SUITES = []
 BRIDGES = []
 
 
 def explore(args):
     explore_firmwares(args)
-    explore_suites()
     explore_bridges()
 
 
-def explore_firmwares(args):
-    scan_glob = "firmware/bin/*"
+def explore_firmwares(args: Any) -> None:
+    firmware_binary_files = str(ROOT_DIR / "firmware/bin/*")
     if args.verbosity > 0:
-        print("Scanning {}".format(colored(scan_glob, "yellow")))
-    for fw in glob.glob(scan_glob):
-        if "trezor-emu-core" in fw:
-            version = fw.replace("firmware/bin/trezor-emu-core-v", "")
+        print(f"Scanning {colored(firmware_binary_files, 'yellow')}")
+
+    # Analyzing all potential emulator files and gathering their versions
+    #   (version is located at the end, after emulator identifier and '-v')
+    identifier_TT = "trezor-emu-core"
+    identifier_T1 = "trezor-emu-legacy"
+    for fw in glob.glob(firmware_binary_files):
+        if identifier_TT in fw:
+            version = fw.split(identifier_TT + "-v")[-1]
             FIRMWARES["TT"].append(version)
-            if args.verbosity > 0:
-                print(
-                    "  found {} => {}".format(
-                        colored(fw, "yellow"), colored(version, "magenta")
-                    )
-                )
-        elif "trezor-emu-legacy" in fw:
-            version = fw.replace("firmware/bin/trezor-emu-legacy-v", "")
+        elif identifier_T1 in fw:
+            version = fw.split(identifier_T1 + "-v")[-1]
             FIRMWARES["T1"].append(version)
-            if args.verbosity > 0:
-                print(
-                    "  found {} => {}".format(
-                        colored(fw, "yellow"), colored(version, "magenta")
-                    )
-                )
         else:
             if args.verbosity > 0:
-                print("  skipping {}".format(colored(fw, "yellow")))
+                print(f"  skipping {colored(fw, 'yellow')}")
+            continue
+
+        if args.verbosity > 0:
+            print(f"  found {colored(fw, 'yellow')} => {colored(version, 'magenta')}")
 
     for model in FIRMWARES.values():
         model.sort(key=sort_firmwares, reverse=True)
 
 
-def sort_firmwares(version):
+def sort_firmwares(version: str) -> tuple:
     if "master" in version:
         return 0, 0, 0
     return tuple(int(n) for n in version.split("."))
 
 
-def explore_suites():
-    # TODO automate
-    SUITES.append("20.8.1")
-    SUITES.append("20.7.1")
-
-
-def explore_bridges():
+def explore_bridges() -> None:
     # TODO automate
```

Comment:
Let's remove the TODO. I do not think it makes sense to automate as there are only a few releases of Bridge per year.

Reply:
Yes, the same "issue" is in the index.html, we can also auto-generate the bridge buttons, but there is no big need